### PR TITLE
WebHost: Set Generator memory limit to 4GiB

### PR DESCRIPTION
**Pull request by black-sliver**
## What is this fixing or adding?

Sets WebHost generator memory limit to 4GB

## How was this tested?

`_ = bytearray(8*1024*1024*1024)` in gen_game()'s task

### CODE REVIEW

**Comment by palex00** [comment]
May I dumbly ask why? The more it can use the faster, no?

**Comment by nicholassaylor** [comment]
The site has maxed out on memory multiple times since 0.5.1 released, meaning that rooms may be unable to start. Do we have any data with how much memory Generator uses currently?

**Comment by black-sliver** [comment]
> May I dumbly ask why? The more it can use the faster, no?

Not really. Generating a seed takes X amount of memory. Throwing more at it does not make it faster unless the code is changed to be "wasteful but faster" (which would break other things anyway).

> Do we have any data with how much memory Generator uses currently?

Not that I know of. There are some options that are not banned from WebHost that use a lot of memory and I hope by allowing 4GB we don't break those, even if all 20 players pick them, while giving the system enough slack to kill misbehaving processes.

**Comment by Berserker66** [comment]
Should probably be a config option and it feels odd to have it in a function called init_db.. doesn't that affect more than gen as well?

**Comment by black-sliver** [comment]
> Should probably be a config option

Done.

> it feels odd to have it in a function called init_db.. doesn't that affect more than gen as well?

This is already done in #4310, however since the two are conflicting anyway (now even more), I pushed the rename here as well.

File: WebHostLib/__init__.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
 app.config["JOB_THRESHOLD"] = 1
 # after what time in seconds should generation be aborted, freeing the queue slot. Can be set to None to disable.
 app.config["JOB_TIME"] = 600
+# memory limit for generator processes
```
Comment (Berserker66) [inline WebHostLib/__init__.py:42]:
```suggestion
# memory limit for generator processes in bytes
```

**Review by Berserker66 — APPROVED** [review]

File: docs/webhost configuration sample.yaml [hunk 2]
```diff
@@ -27,8 +27,14 @@
 # If you wish to deploy, uncomment the following line and set it to something not easily guessable.
 # SECRET_KEY: "Your secret key here"
 
-# TODO
-#JOB_THRESHOLD: 2
+# Slot limit to post a generation to Generator process pool instead of rolling directly in WebHost process
+#JOB_THRESHOLD: 1
+
+# After what time in seconds should generation be aborted, freeing the queue slot. Can be set to None to disable.
+#JOB_TIME = 600
```
Comment (black-sliver) [inline docs/webhost configuration sample.yaml:34]:
```suggestion
#JOB_TIME: 600
```

**Pull request merged**